support-workers/src/typescript/model/address.ts

```diff
@@ -0,0 +1,108 @@
+import { z } from 'zod';
```

Comment:
I find it quite hard to follow this code, probably partly because I'm not used to the language, but also it's not really clear what the aim is.  I'm not sure if the naming could be improved, and perhaps some comments?